[Actions] Converted `rejectUnauthorized` config usages to `verificationMode`. (#100179)
elastic/kibana: docs/settings/alert-action-settings.asciidoc

  The options `smtp.ignoreTLS` and `smtp.requireTLS` can not both be set to true.

| `xpack.actions.customHostSettings[n]`

> `.tls.rejectUnauthorized` [Deprecated. Use `xpack.actions.customHostSettings.tls.verificationMode`] {ess-icon}

Reviewer: ```suggestion
`.tls.rejectUnauthorized` {ess-icon}
```

>   | A boolean value indicating whether to bypass server certificate validation.

Reviewer: ```suggestion
  | *Deprecated. Use `xpack.actions.customHostSettings.tls.verificationMode` instead.* A boolean value indicating whether to bypass server certificate validation.
```

  Overrides the general `xpack.actions.rejectUnauthorized` configuration
  for requests made for this hostname/port.

| `xpack.actions.customHostSettings[n]`
`.tls.verificationMode`

>   | Controls the verification of the server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection to {es}. Valid values are "`full`", "`certificate`", and "`none`". 

Reviewer: ```suggestion
  | Controls the verification of the server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection to {es}. Valid values are `full`, `certificate`, and `none`. 
```

>   Using "`full`" performs hostname verification, using "`certificate`" skips hostname verification, and using "`none`" skips verification entirely. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.

Reviewer: ```suggestion
 Use `full` to perform hostname verification, `certificate` to skip hostname verification, and `none` to skip verification. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.
```

  | Specifies HTTP headers for the proxy, if using a proxy for actions. Defaults to {}.

a|`xpack.actions.`

> `proxyRejectUnauthorizedCertificates` [Deprecated Use `xpack.actions.tls.proxyVerificationMode`.] {ess-icon}

Reviewer: ```suggestion
`proxyRejectUnauthorizedCertificates` {ess-icon}
```

>   | Set to `false` to bypass certificate validation for the proxy, if using a proxy for actions. Defaults to `true`.

Reviewer: ```suggestion
  | *Deprecated. Use `xpack.actions.tls.proxyVerificationMode` instead.*] Set to `false` to bypass certificate validation for the proxy, if using a proxy for actions. Defaults to `true`.
```

a|`xpack.actions[n]`
`.tls.proxyVerificationMode` {ess-icon}

> | Controls the verification for the proxy server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection to {es}. Valid values are "`full`", "`certificate`", and "`none`". 

Reviewer: ```suggestion
| Controls the verification for the proxy server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection to {es}. Valid values are `full`, `certificate`, and `none`. 
```

>   Using "`full`" performs hostname verification, using "`certificate`" skips hostname verification, and using "`none`" skips verification entirely. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.

Reviewer: ```suggestion
Use `full` to perform hostname verification, `certificate` to skip hostname verification, and `none` to skip verification. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.
```

> | `xpack.actions.rejectUnauthorized` [Deprecated. Use `xpack.actions.tls.verificationMode`.] {ess-icon}

Reviewer: ```suggestion
| `xpack.actions.rejectUnauthorized` {ess-icon}
```

>   | Set to `false` to bypass certificate validation for actions. Defaults to `true`. +

Reviewer: ```suggestion
  | *Deprecated. Use `xpack.actions.tls.verificationMode` instead*.] Set to `false` to bypass certificate validation for actions. Defaults to `true`. +
```

  +
  As an alternative to setting `xpack.actions.rejectUnauthorized`, you can use the setting
  `xpack.actions.customHostSettings` to set TLS options for specific servers.

a|`xpack.actions[n]`
`.tls.verificationMode` {ess-icon}

> | Controls the verification for the server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection to {es}. Valid values are "`full`", "`certificate`", and "`none`". 

Reviewer: ```suggestion
| Controls the verification for the server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection to {es}. Valid values are `full`, `certificate`, and `none`. 
```

>   Using "`full`" performs hostname verification, using "`certificate`" skips hostname verification, and using "`none`" skips verification entirely. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.

Reviewer: ```suggestion
  Use `full` to perform hostname verification, `certificate` to skip hostname verification, and `none` to skip verification. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.
```

a|`xpack.actions.`
`proxyRejectUnauthorizedCertificates` {ess-icon}

>   | *Deprecated. Use `xpack.actions.tls.proxyVerificationMode` instead.*] Set to `false` to bypass certificate validation for the proxy, if using a proxy for actions. Defaults to `true`.

Reviewer: ```suggestion
  | *Deprecated. Use `xpack.actions.tls.proxyVerificationMode` instead.* Set to `false` to bypass certificate validation for the proxy, if using a proxy for actions. Defaults to `true`.
```

a|`xpack.actions[n]`
`.tls.proxyVerificationMode` {ess-icon}
| Controls the verification for the proxy server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection to the proxy server. Valid values are `full`, `certificate`, and `none`. 
Use `full` to perform hostname verification, `certificate` to skip hostname verification, and `none` to skip verification. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.

| `xpack.actions.rejectUnauthorized` {ess-icon}

>   | *Deprecated. Use `xpack.actions.tls.verificationMode` instead*.] Set to `false` to bypass certificate validation for actions. Defaults to `true`. +

Reviewer: ```suggestion
  | *Deprecated. Use `xpack.actions.tls.verificationMode` instead*. Set to `false` to bypass certificate validation for actions. Defaults to `true`. +
```

  +
  As an alternative to setting `xpack.actions.rejectUnauthorized`, you can use the setting
  `xpack.actions.customHostSettings` to set TLS options for specific servers.

a|`xpack.actions[n]`
`.tls.verificationMode` {ess-icon}
| Controls the verification for the server certificate that {hosted-ems} receives when making an outbound SSL/TLS connection for actions. Valid values are `full`, `certificate`, and `none`. 
  Use `full` to perform hostname verification, `certificate` to skip hostname verification, and `none` to skip verification. *Default: `full`*. <<elasticsearch-ssl-verificationMode,Equivalent {kib} setting>>.
  +

>   As an alternative to setting `xpack.actions.tls.verificationMode`, you can use the setting

Reviewer: Add a space above this line.
